subsys/net/ip/net_if.c

```diff
@@ -4654,6 +4661,7 @@ static void notify_iface_up(struct net_if *iface)
 	}
 
 	net_if_flag_set(iface, NET_IF_RUNNING);
+	rejoin_multicast_groups(iface);
```

Comment:
Should we actually move the
```
	if (!net_if_is_offloaded(iface)) {
		/* Make sure that we update the IPv6 addresses and join the
		 * multicast groups.
		 */
		rejoin_multicast_groups(iface);
		net_if_start_dad(iface);
	}
```
from `net_if_up()` to here. There is the point-to-point check few lines below where these two statements could be placed. I am not sure we need to do the rejoining in two places.

Reply:
Ok, thanks for the hints. Now the rejoin_multicast is called only if the operational state is NET_IF_OPER_UP:

`	if (net_if_oper_state(iface) == NET_IF_OPER_UP) {
		if (prev_state != NET_IF_OPER_UP) {
			notify_iface_up(iface);
		}`
		
I presume that's ok?